tests/store/artifact/utils/test_model_utils.py

```diff
@@ -33,6 +33,12 @@ def test_parse_models_uri_with_version(uri, expected_name, expected_version):
         ("models:/AdsModel1/pROduction", "AdsModel1", "pROduction"),  # case insensitive
         ("models:/Ads Model 1/None", "Ads Model 1", "None"),
         ("models://scope:key@databricks/Ads Model 1/None", "Ads Model 1", "None"),
+        (
+            "models:/Name/Stage@Alias",
+            "Name",
+            "Stage@Alias",
+        ),  # technically allowed, but the backend would throw
+        ("models:/Name@Alias/Stage", "Name@Alias", "Stage"),
```

Comment:
Can we add just one additional test with some wacky naming conventions in the model name so that we're future-protected in case (when) someone modifies this logic next? 
`"models:/A#!&$%;{}()[]CrazyName@Alias"` or something like that?

Reply:
^ Huge +1!

Reply:
Looks like that test case currently fails since the parser only considers the path till `#`

Reply:
We even have a failure case actually `"models://Name@te#ty;",  # invalid characters`

Reply:
When was that failure case added? (Making sure that this was a preexisting limitation and not something new)

Reply:
Good call out, looks like I added it recently 😂 : https://github.com/mlflow/mlflow/commit/93ee3525205ab217db7261552d3190d4bf02f371#diff-1e7275af9760961835ec5188323cbee64e47d10cec7e76632ab54224221f768aR98. How should we parse in `#` though? I know @BenWilson2 had some struggles with it

Reply:
Can we `allow_fragments=True`

Reply:
Yup that worked! Great call @harupy